api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/RegisterBabyAction.java

```diff
@@ -0,0 +1,192 @@
+            for (Obs candidate : encounter.getObsAtTopLevel(false)) {
+                if (candidate.getConcept().equals(newbornDetailsConcept)) {
+                    Set<Obs> groupMembers = candidate.getGroupMembers();
+                    if ( groupMembers != null ) {
+                        for (Obs groupMember : groupMembers) {
+                            // search for Labour and Delivery obs that have the "Register patient in EMR" obs set to No (SL-617)
+                            if (groupMember.getConcept().equals(registerBabyConcept) && groupMember.getValueCoded().equals(noConcept)) {
+                                //find the gender and the birthdate time of the baby
+                                String gender = getObsValue(groupMembers, BABY_GENDER_CONCEPT_SOURCE, BABY_GENDER_CONCEPT);
+                                Date birthDatetime = getObsDateValue(groupMembers, DATE_TIME_OF_BIRTH_CONCEPT_SOURCE, DATE_TIME_OF_BIRTH_CONCEPT);
+                                if (StringUtils.isNotBlank(gender) && (birthDatetime != null)) {
+                                    Patient baby = registerBaby(patient, gender, birthDatetime, encounter.getLocation());
+                                    if (baby != null && baby.getUuid() != null) {
+                                        groupMember.setValueCoded(yesConcept);
+                                        groupMember.setComment(baby.getUuid());
+                                        updateEncounter = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            if (updateEncounter) {
+                Context.getEncounterService().saveEncounter(encounter);
+            }
+
+            //automatically redirect to the children page
+            formEntrySession.setAfterSaveUrlTemplate("pihcore/children/children.page?patientId={{patient.id}}");
+        }
+    }
+
+    Patient registerBaby(Patient mother, String gender, Date birthDatetime, Location location) {
+        PersonService personService = Context.getPersonService();
+        RelationshipType motherToChildRelationshipType = personService.getRelationshipTypeByUuid(PihEmrConfigConstants.RELATIONSHIPTYPE_MOTHERTOCHILD_UUID);
+        RelationshipType relationship = personService.getRelationshipTypeByUuid(motherToChildRelationshipType.getUuid());
+        List<Relationship> relationships = new ArrayList<Relationship>();
+        if (relationship !=null) {
+            relationships.add(new Relationship(mother, null, motherToChildRelationshipType));
+        }
+        Patient baby = new Patient();
+        PersonName babyName = new PersonName();
+        babyName.setFamilyName(mother.getFamilyName());
+        babyName.setGivenName("Baby");
+        baby.addName(babyName);
+        String babyGender = gender;
+        if (StringUtils.equalsIgnoreCase(gender, "Female")) {
```

Comment:
Can we change to match directly on concept instead of valueAsString?  (I think valueAsString is locale/language specific, among other things)?

Reply:
that is what I initially did, but the concept used here https://kgh-test.pih-emr.org/openmrs/dictionary/concept.htm?conceptId=3253 returns "Male". If we store "Male" as gender in the db, that would be different than all other records,  which are stored as "M".
Would that be OK to store "Male" as a gender in the db?

Reply:
What @mogoodrich is talking about is line 68 @cioan , where you get "Male" as a string obs value, rather than getting `3ce84b1c-26fe-102b-80cb-0017a47871b2` as the obs value coded concept id.  Please do that instead.

Reply:
Yes, correct, @mseaton

Reply:
np, I updated the code